x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -493,131 +549,139 @@ public void runAsDenied(Authentication authentication, String action, TransportM
-                } else {
-                    logger.info("{}[transport] [run_as_denied]\t{}, {}, roles=[{}], action=[{}], request=[{}]{}",
-                            localNodeInfo.prefix, originAttributes(threadContext, message, localNodeInfo), runAsSubject(authentication),
-                            arrayToCommaDelimitedString(roleNames), action, message.getClass().getSimpleName(), opaqueId());
+                    logEntry.with(INDICES_FIELD_NAME, Strings.arrayToCommaDelimitedString(indices.get()));
                 }
+                opaqueId(threadContext, logEntry);
+                logger.info(logEntry);
             }
         }
     }
 
     @Override
     public void runAsDenied(Authentication authentication, RestRequest request, String[] roleNames) {
         if (events.contains(RUN_AS_DENIED)
-                && (eventFilterPolicyRegistry.ignorePredicate().test(new AuditEventMetaInfo(Optional.of(authentication.getUser()),
-                        Optional.of(effectiveRealmName(authentication)), Optional.of(roleNames), Optional.empty())) == false)) {
-            if (includeRequestBody) {
-                logger.info("{}[rest] [run_as_denied]\t{}, {}, roles=[{}], uri=[{}], request_body=[{}]{}",
-                        localNodeInfo.prefix, hostAttributes(request), runAsSubject(authentication),
-                        arrayToCommaDelimitedString(roleNames), request.uri(), restRequestContent(request), opaqueId());
-            } else {
-                logger.info("{}[rest] [run_as_denied]\t{}, {}, roles=[{}], uri=[{}]{}", localNodeInfo.prefix,
-                        hostAttributes(request), runAsSubject(authentication), arrayToCommaDelimitedString(roleNames), request.uri(),
-                        opaqueId());
+                && eventFilterPolicyRegistry.ignorePredicate().test(new AuditEventMetaInfo(Optional.of(authentication.getUser()),
+                        Optional.of(effectiveRealmName(authentication)), Optional.of(roleNames), Optional.empty())) == false) {
+            final StringMapMessage logEntry = new StringMapMessage(this.entryCommonFields.commonFields);
+            logEntry.with(EVENT_TYPE_FIELD_NAME, REST_ORIGIN_FIELD_VALUE)
+                    .with(EVENT_ACTION_FIELD_NAME, "run_as_denied")
+                    .with(PRINCIPAL_ROLES_FIELD_NAME, arrayToCommaDelimitedString(roleNames));
+            restUri(request, logEntry);
+            runAsSubject(authentication, logEntry);
+            restOrigin(request, logEntry);
+            requestBody(request, logEntry);
+            opaqueId(threadContext, logEntry);
+            logger.info(logEntry);
+        }
+    }
+
+    private void requestBody(RestRequest request, StringMapMessage logEntry) {
+        if (includeRequestBody) {
+            final String requestContent = restRequestContent(request);
+            if (Strings.hasLength(requestContent)) {
+                logEntry.with(REQUEST_BODY_FIELD_NAME, requestContent);
```

Comment:
Adds the `request.body` only if enabled and the request content is non-empty.
The request content is escaped in the `log4j2.properties` format file (see [above](https://github.com/elastic/elasticsearch/pull/31931/files#r209964478)), it does not have to be escaped here.